server/s3_put_bucket_replication_action.cc

```diff
@@ -0,0 +1,213 @@
+        );
+  }
+  s3_log(S3_LOG_DEBUG, "", "%s Exit", __func__);
+}
+
+void S3PutBucketReplicationAction::consume_incoming_content() {
+  s3_log(S3_LOG_INFO, stripped_request_id, "Consume data\n");
+  if (request->is_s3_client_read_error()) {
+    client_read_error();
+  } else if (request->has_all_body_content()) {
+    new_bucket_replication_content = request->get_full_body_content_as_string();
+    validate_request_body(new_bucket_replication_content);
+  } else {
+    // else just wait till entire body arrives. rare.
+    request->resume();
+  }
+  s3_log(S3_LOG_DEBUG, "", "%s Exit", __func__);
+}
+
+void S3PutBucketReplicationAction::validate_request_body(std::string content) {
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  put_bucket_replication_body =
+      put_bucket_replication_body_factory->create_put_replication_body(
+          content, request_id);
+  if (put_bucket_replication_body->isOK()) {
+
+    replication_config_json =
+        put_bucket_replication_body->get_replication_configuration_as_json();
+
+    if (!replication_config_json.empty())
+      s3_log(S3_LOG_INFO, stripped_request_id, "replication_config_json = %s\n",
+             replication_config_json.c_str());
+
+    next();
+  } else {
+    std::string s3_error =
+        put_bucket_replication_body->get_additional_error_information();
+    if (s3_error.compare("InvalidTag") == 0) {
+      set_s3_error("InvalidTag");
+      send_response_to_s3_client();
+    } else if (s3_error.compare("InvalidRequestForFilter") == 0) {
+      set_s3_error("InvalidRequestForFilter");
+      send_response_to_s3_client();
+
+    } else if (s3_error.compare("InvalidRequestForTagFilter") == 0) {
+      set_s3_error("InvalidRequestForTagFilter");
+      send_response_to_s3_client();
+
+    } else if (s3_error.compare("InvalidArgumentDuplicateRulePriority") == 0) {
+      set_s3_error("InvalidArgumentDuplicateRulePriority");
+      send_response_to_s3_client();
+
+    } else if (s3_error.compare("InvalidArgumentDuplicateRuleID") == 0) {
+      set_s3_error("InvalidArgumentDuplicateRuleID");
+      send_response_to_s3_client();
+    } else {
+      set_s3_error("MalformedXML");
+      send_response_to_s3_client();
+    }
+  }
+  s3_log(S3_LOG_DEBUG, "", "%s Exit", __func__);
+}
+
+void S3PutBucketReplicationAction::fetch_bucket_info_failed() {
+  s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  if (bucket_metadata->get_state() == S3BucketMetadataState::missing) {
+    set_s3_error("NoSuchBucket");
+  } else if (bucket_metadata->get_state() ==
+             S3BucketMetadataState::failed_to_launch) {
+    s3_log(S3_LOG_ERROR, request_id,
+           "Bucket metadata load operation failed due to pre launch failure\n");
+    set_s3_error("ServiceUnavailable");
+  } else {
+    set_s3_error("InternalError");
+  }
+  send_response_to_s3_client();
+  s3_log(S3_LOG_DEBUG, "", "%s Exit", __func__);
+}
+
+void
```

Comment:
Why is the return and the function name on different lines here. That isn't the normal style.

Reply:
We format the code using clang-format -style and if the function name is big, while formatting using command - clang-format -style=Google -i <filename> it does this automatically. If we try to adjust it, it gives formatting error.